src/main/java/org/myrobotlab/framework/interfaces/ServiceInterface.java

```diff
@@ -162,4 +172,55 @@
 
   public int getCreationOrder();
 
+  static public LinkedHashMap<String, ServiceConfig> getDefault(String name, String type) {

```

Comment:
Why do you need to pass Type here?  the service interface contains getType()  ...  you should be able to make that static and use it here also to avoid needing to pass the type in.

Reply:
i stand semi-corrected...  ignore this.

Reply:
actually.. don't ignore this.. you can derive the type of the implementing class with
`public static String getType() {
    return MethodHandles.lookup().lookupClass().getSimpleName();
  };`